database/sql/V048__create_settlement_rewards.sql

```diff
@@ -0,0 +1,31 @@
+-- Populated before settlement occurs on chain
+-- (after the submissions have been ranked)
+CREATE TABLE settlement_scores (
+  transaction_id bytea NOT NULL,
+  winning_score numeric(78,0) NOT NULL,
+  reference_score numeric(78,0) NOT NULL
+);
+
+-- Populated after block finalization via transactionReceipt.
+CREATE TABLE settlement_observations (
+  -- the appended transaction identifier read from the call data
+  transaction_id bytea NOT NULL,
+  -- block number and log index to uniquely `JOIN` on the `settlements`
+  -- table, read from the transaction receipt
+  block_number bigint NOT NULL,
+  log_index bigint NOT NULL,
+  gas_used numeric(78,0) NOT NULL,
+  effective_gas_price numeric(78,0) NOT NULL,
+  -- the surplus observed from the transaction call data,
+  -- and converted to ETH with the auction external prices.
+  surplus numeric(78,0) NOT NULL,
+  fee numeric(78,0) NOT NULL,
+
+  PRIMARY KEY (block_number, log_index)
```

Comment:
I think that the primary key should be `settlement_id`. I think we should add an index on `(block_number, log_index)` and that they should be unique, but I don't agree that it the primary key.

Reply:
I suggested this as the primary key before. Note that `settlement_id` (now called `transaction_id`) can't be the primary key because we cannot control whether some onchain settlement transaction reuses an id.